projects/validation-messages/src/lib/services/validation-messages.service.ts

```diff
@@ -42,46 +44,28 @@ export class ValidationMessagesService {
 
   /**
    * Set validation messages.
-   * @param {ValidationMessages} validationMessages
+   * @param {ValidationMessage} validationMessages
    * @param patternValidationMessages
    */
   static setValidationMessages(
-    validationMessages: ValidationMessages,
+    validationMessages: ValidationMessagesConfig,
     patternValidationMessages: any = {}
   ): void {
+    ValidationMessagesService.validationMessages = {};
```

Comment:
first create an object that will be hydrated and then at the end assign that object to the service